Jetpack Agency Dashboard - Plugin Management: add plugin management multi-site routes and component

yashwin opened the pull request:
#### Proposed Changes

This PR adds component and routes for the multi-site view behind a feature flag for the Plugin Management

#### Testing Instructions

**Prerequisites**

Since this change is made specifically for agencies, you must set yourself(partner) as an agency - 2c49b-pb. Make sure to switch it back to the previous type.

**Instructions**

1. Run `git checkout add/plugin-management-routes-in-jetpack-cloud` and `yarn start-jetpack-cloud`
2. Open http://jetpack.cloud.localhost:3000/, and you'll be redirected to the /dashboard.
3. Verify that a new sidebar navigation item `Plugins` is shown in the Dashboard as shown below with a placeholder text.

<img width="613" alt="Screenshot 2022-07-19 at 12 11 42 PM" src="https://user-images.githubusercontent.com/10586875/179683320-51cc8633-2448-43d0-bcd8-9137208ad46d.png">

4. Verify that the feature flag **jetpack/plugin-management** is set to true only in dev environments.
5. Switch to the original partner type - 2c49b-pb and verify that you are redirected to /landing when you visit http://jetpack.cloud.localhost:3000/dashboard/plugins/manage.
6. Switch the partner to agency and set the feature flag **jetpack/plugin-management** to false in `config/development.json` and `config/jetpack-cloud-development.json` and restart the server `yarn start-jetpack-cloud` and verify that **Plugins** navigation is not shown & verify that you are redirected to /landing when you visit http://jetpack.cloud.localhost:3000/dashboard/plugins/manage.
7. Verify that section `jetpack-cloud-plugin-management` is enabled only in `config/jetpack-cloud-development.json`

#### Pre-merge Checklist

Complete applicable items on this checklist **before** merging into the trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.

- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes? Will be added later - 1202518759611394-as-1202627702018877
- [x] ~~Have you tested the feature in [Simple](P9HQHe-k8-p2-p2), [Atomic](P9HQHe-jW-p2-p2), and [self-hosted Jetpack sites](PCYsg-g6b-p2-p2)?~~
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Are we memoizing when appropriate (for expensive computations)? More info in [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md) and [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors)
- [x] Have we sent any new strings [for translation](PCYsg-1vr-p2-p2) ASAP?

Related to 1202518759611394-as-1202533324413691

yashwin committed:
Add plugin management multi-site routes

manzoorwanijk commented on client/jetpack-cloud/sections/agency-dashboard/controller.tsx:
```diff
@@ -16,3 +17,10 @@ export function agencyDashboardContext( context: PageJS.Context, next: () => voi
 	next();
 }
+
+export function pluginManagementContext( context: PageJS.Context, next: () => void ): void {
```
You can use the in-built `VoidFunction`
```suggestion
export function pluginManagementContext( context: PageJS.Context, next: VoidFunction ): void {
```

yashwin replied:
Thanks for the review, I have made the changes.

manzoorwanijk commented on client/jetpack-cloud/sections/agency-dashboard/plugin-management/index.tsx:
```diff
@@ -0,0 +1,43 @@
+	}
+
+	if ( hasFetched ) {
+		return <>Plugin Management</>;
```
Doesn't this need to be translated?

yashwin replied:
No, this is a placeholder that will be removed in the upcoming PR.

yashwin committed:
Use in-built VoidFunction

rcanepa commented:
Yashwin, as you might know, the plugins section won't be part of the dashboard (even if they are linked), so we need to remove `dashboard` from the path to the URL.

yashwin committed:
Use /plugins as URL

Also, remove condition to check if plugin management is enabled since the component handles this scenario

yashwin committed:
Move plugin management into new module and add sections

yashwin commented:
> Yashwin, as you might know, the plugins section won't be part of the dashboard (even if they are linked), so we need to remove `dashboard` from the path to the URL.

@rcanepa - That makes total sense. I have made the changes. I have also moved plugin management into its own module since it makes more sense to keep them separate.

rcanepa commented on config/jetpack-cloud-horizon.json:
```diff
@@ -60,6 +60,7 @@
 		"backup": true,
 		"jetpack-cloud": true,
 		"jetpack-cloud-agency-dashboard": true,
+		"jetpack-cloud-plugin-management": false,
```
Let's make this one true since I believe this is the env behind calipso.live URLs.

rcanepa reviewed (approved):
LGTM!
